### PR TITLE
ListView: Orientation change support

### PR DESCRIPTION
**Pull request by ldanilina**
Portrait layouts:
- TopVertical
- TopHorizontal
- BottomVertical
- BottomHorizontal

Landscape layouts:
- LeadingVertical
- TrailingVertical

### CODE REVIEW

File: LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSCore/Utils/DeviceRotationViewModifier.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
+//
+//  DeviceRotationViewModifier.swift
+//
+
+import UIKit
+import SwiftUI
+
+public struct DeviceRotationViewModifier: ViewModifier {
```
Comment (aravind-raveendran) [inline LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSCore/Utils/DeviceRotationViewModifier.swift:8]: A good place for this ViewModifier is the DolbyIORTSUIKit framework or even the DolbyIOUIKit ?

File: LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSCore/Utils/DeviceRotationViewModifier.swift [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    let action: (UIDeviceOrientation) -> Void
+
+    public func body(content: Content) -> some View {
+        content
+            .onAppear {
+                guard let interfaceOrientation = UIApplication.shared.connectedScenes.map({ $0 as? UIWindowScene }).compactMap({ $0 }).first?.windows.first?.windowScene?.interfaceOrientation else {
+                    action(UIDevice.current.orientation)
+                    return
+                }
+                action(UIDeviceOrientation(rawValue: (interfaceOrientation).rawValue)!)
```
Comment (aravind-raveendran) [inline LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSCore/Utils/DeviceRotationViewModifier.swift:18]: Can we also move `UIDeviceOrientation(rawValue: (interfaceOrientation).rawValue)` to the `guard let` statement ?

File: LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSCore/Utils/DeviceRotationViewModifier.swift (hunk 1 above)
Comment (ldanilina) [reply LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSCore/Utils/DeviceRotationViewModifier.swift]: Thanks a lot @aravind-raveendran ! addressed

File: LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSCore/Utils/DeviceRotationViewModifier.swift (hunk 2 above)
Comment (ldanilina) [reply LocalPackages/DolbyIORTSUIKit/Sources/DolbyIORTSCore/Utils/DeviceRotationViewModifier.swift]: done

**Comment by CasperMok** [comment]
Upside down doesn't work properly

**Comment by CasperMok** [comment]
May I suggest the following for the ListViewLayout for your consideration? Don't have to do it now.
This shall cover all the 8 cases we need in the future. For those, not supporting now, we may just return error/fatal for the moment. 

enum Scrolling {
case horizontal(col: Int)
case vertical(row: Int)
}

enum ListViewLayout {
case top(Scrolling)
case bottom(Scrolling)
case leading(Scrolling)
case trailing(Srolling)
}

**Comment by ldanilina** [comment]
> May I suggest the following for the ListViewLayout for your consideration? Don't have to do it now. This shall cover all the 8 cases we need in the future. For those, not supporting now, we may just return error/fatal for the moment.
> 
> enum Scrolling { case horizontal(col: Int) case vertical(row: Int) }
> 
> enum ListViewLayout { case top(Scrolling) case bottom(Scrolling) case leading(Scrolling) case trailing(Srolling) }

Thanks Casper.
Sure, we can do it that way, but that would generate some modes that don't really make sense (trailing horizontal, leading horizontal). This is out of scope for this ticket, so we can discuss further

**Comment by ldanilina** [comment]
> Upside down doesn't work properly Portrait -> Landscape Left/Right -> Upside down
> 
fixed now

**Review by CasperMok — APPROVED** [review]

**Pull request merged**